ttnn/cpp/ttnn/operations/moreh/moreh_sgd/moreh_sgd.hpp

```diff
@@ -0,0 +1,36 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+
+namespace ttnn::operations::moreh::moreh_sgd {
+struct MorehSgd {
+    static std::vector<std::optional<Tensor>> invoke(
+        const Tensor& param_in,
+        const Tensor& grad,
+        const std::optional<const Tensor> momentum_buffer_in,
+        const std::optional<const Tensor> param_out,
+        const std::optional<const Tensor> momentum_buffer_out,
```

Comment:
Use const std::optional<Tensor>&

Reply:
resolved in [a981edd](https://github.com/tenstorrent/tt-metal/pull/12378/commits/a981edd24c59337a422bc4c59ae88b14489c75a5)